app/v2/controllers/V2MovementsController.scala

```diff
@@ -582,6 +612,51 @@ class V2MovementsControllerImpl @Inject() (
         } yield result
     }
 
+  private def streamLargeMessage(
+    eori: EORINumber,
+    movementId: MovementId,
+    messageId: MessageId,
+    messageTypeList: Seq[MessageType],
+    src: Source[ByteString, _],
+    movementType: MovementType
+  )(implicit hc: HeaderCarrier) =
+    withReusableSource(src) {
+      source =>
+        for {
+
+          messageType <- xmlParsingService.extractMessageType(source, messageTypeList).asPresentation
+          messageUpdate          = MessageUpdate(_, None, Some(messageType))
+          updateMovementResponse = persistenceService.updateMessage(eori, movementType, movementId, messageId, _)
+          _ <- updateMovementResponse(messageUpdate(MessageStatus.Processing)).asPresentation
+          _ <- validationService
+            .validateXml(messageType, source)
+            .asPresentation
+            .leftMap {
+              err =>
+                updateMovementResponse(messageUpdate(MessageStatus.Failed)).value
+                err
+            }
+          _ <- persistenceService.updateMessageBody(eori, movementType, movementId, messageId, source).asPresentation
```

Comment:
@drnaylor-oc  With this newly created endpoint to update the body I can see we are checking the status on transit-movements code and if that status is Pending then only it is going to update the body otherwise it will return error
https://github.com/hmrc/transit-movements/blob/main/app/uk/gov/hmrc/transitmovements/controllers/MessageBodyController.scala#L115
Now here at this point of time we will having message status as Processing hence will be getting error from transit-movements.  So either Api or Transit-movements we need to modify the code. Please suggest.

Reply:
The only time that a body should not exist is if the status is `Pending`. It should be immutable after that. So, if something is setting the status to `Processing` before uploading the body, then that's wrong -- hence, the API needs to change.

Reply:
@olakunlekuye2  in that case you are going to set message status as Processing post this call. Currently you are setting it before making to this call. Just shuffle that code and then I hope it should be fine.

Reply:
You don't need to set the status at all, that call will do it for you.

Reply:
Done